File: Ukol_1.m
% % % Skript pro výpočet změny U, H, S, F a G pro 1 mol chlóru

% % Nadefinované podmínky
T1 = 298; % K
p1 = 01*10^6; % Pa

T2 = 500; % K
p2 = 10*10^6; % Pa

% % Tabelované hodnoty a funkce (webbook.nist.gov/cgi/cbook.cgi?ID=C7782505&Mask=1)
% (použitelné pouze v rozmezí 298-1000 K)
% Konstanty:
A = 33.05060;
B = 12.22940;
C = -12.06510;
D = 4.385330;
E = -0.159494;
G = 259.0290;

% Funkce
% St = @(t) A*log(t) + B*t + (C*t^2)/2 + (D*t^3)/3 -E/(2*t^2) + G;
Cpot = @(t) A + B*t + C*t.^2 + D*t.^3 - E./(t.^2);
% S = @(t) Sot(t/1000); % J/(mol*K)
Cpo = @(t) Cpot(t/1000); % J/(mol*K)

% Hodnoty
So = 223.08; % J/(mol*K)
S1 = So;

% % Mezivýpočty
% Přepočet Cp na Cv
R = 8314;
Cvo = @(t) Cpo(t) - R;

% Výpočet V1 a V2
V1 = R*T1/p1;
V2 = R*T2/p2;

% % Výpočet delS
funkce_ = @(t) Cpo(t)./t;
S2 = S1 + 0 + integral(funkce_, T1, T2) - R*log(p2/p1) + 0; % 0 znamenají nulové integrály R/p* - R/p, protože počítáme plyn ve stavu IP
delS = S2 - S1;

% % Výpočet delU
funkce_ = @(t) Cvo(t);
delU = 0 + integral(funkce_, T1, T2) + 0;

% % Výpočet delH
funkce_ = @(t) Cpo(t);
delH = 0 + integral(funkce_, T1, T2) + 0;

% % Výpočet delG
delG = delH - T2*S2 + T1*S1;

% % Výpočet delF
delF = delU - T2*S2 + T1*S1;

% Výsledné výpočty
delS
fprintf('   J/(mol*K)\n')
delU
fprintf('   J\n')
delH
fprintf('   J\n')
delG
fprintf('   J\n')
delF
fprintf('   J\n')

File: Ukol_1.py
### Skript pro výpočet změny S, U, H, G a F jednoho molu plynného chlóru
from sympy import *
T = Symbol('T')
R = 8.314

## Nadefinované podmínky
T1 = 298 # K
p1 = 0.1 * 10**6 # Pa

T2 = 500 # K
p2 = 10 * 10**6 # Pa

## Tabulkové hodnoty (Chase, 1998; NIST, 2018)
S1 = 223.08 # J/(mol*K)

A = 33.0506
B = 12.2294
C = -12.0561
D = 4.38533
E = -0.159494

## Mezivýpočet tepelných kapacit
Cp = A + B*T + C*(T**2) + D*(T**3) + E/(T**2)
Cp = Cp.subs(T, T/1000)
Cv = Cp - R

## Výpočet změny entropie
S2 = S1 + integrate(Cp/T, (T, T1, T2)) + R * log(p2/p1)
delS = S2 - S1

## Výpočet změny vnitřní energie
delU = integrate(Cv, (T, T1, T2))

## Výpočet změny entalpie
delH = integrate(Cp, (T, T1, T2))

## Výpočet změny Gibbsovy volné energie
delG = delH - T2*S2 + T1*S1

## Výpočet změny Helmholtzovy volné energie
delF = delU - T2*S2 + T1*S1

## Výsledné změny termodynamických veličin
print('delS = ' + str(N(delS)) + ' J/(mol*K)')
print('delU = ' + str(N(delU)) + ' J/mol')
print('delH = ' + str(N(delH)) + ' J/mol')
print('delG = ' + str(N(delG)) + ' J/mol')
print('delF = ' + str(N(delF)) + ' J/mol')
